packages/eui/src/components/date_picker/react-datepicker/src/calendar.js

```diff
@@ -352,15 +353,29 @@ export default class Calendar extends React.Component {
         </div>
       );
     }
+
     return dayNames.concat(
       [0, 1, 2, 3, 4, 5, 6].map(offset => {
         const day = addDays(cloneDate(startOfWeek), offset);
         const localeData = getLocaleData(day);
+        const localeStartDayOfWeek = localeData._week.dow;
+
+        const weekStartDayIndex = localeStartDayOfWeek + offset;
+        // determine the current day index based on the week starting day (Sunday vs Monday)
+        const currentDayIndex =
+          weekStartDayIndex < 7 ? weekStartDayIndex : 7 - weekStartDayIndex;
+
         const weekDayName = this.formatWeekday(localeData, day);
+        const weekDayNameLong = localeData._weekdays[currentDayIndex]
```

Comment:
```suggestion
        const day = addDays(cloneDate(startOfWeek), offset);
        const localeData = getLocaleData(day);
        const weekDayName = this.formatWeekday(localeData, day);
        const weekDayNameLong = localeData.weekdays(day);
```

The API I'm suggesting, for conciseness

Reply:
Thanks for the suggestion! Updated in this [commit](https://github.com/elastic/eui/pull/7748/commits/52fdd83d1a979366793ad4b10e15ffbbbe9cab7b)